### PR TITLE
39193 Added Ajax search module and lesson model

### PR DESCRIPTION
**Pull request by mmkarim**

### CODE REVIEW

File: app/views/words/show.html.erb [hunk 1]
```diff
@@ -1,21 +1,29 @@
...
-    <% end %>
-  </div>
+<div class="form-inline">
+  <%= form_tag words_path, remote: true, :method => "get"  do  %>
+    <div class="col-md-2">
+      <%= select_tag :search, options_from_collection_for_select(@categories, "id", "title") %>
+    </div>
+    <div class="form-group">
+      <%= radio_button_tag(:option, "learned") %>
+      <%= label_tag(:learned, "Learned") %>
```
Comment (chuanhtuan) [inline app/views/words/show.html.erb]: i18n

File: app/views/words/show.html.erb [hunk 2]
```diff
@@ -1,21 +1,29 @@
...
-    <%= submit_tag :Search, :name => nil, class: "btn btn-primary" %>
-    <% end %>
-  </div>
+<div class="form-inline">
+  <%= form_tag words_path, remote: true, :method => "get"  do  %>
+    <div class="col-md-2">
+      <%= select_tag :search, options_from_collection_for_select(@categories, "id", "title") %>
+    </div>
+    <div class="form-group">
+      <%= radio_button_tag(:option, "learned") %>
```
Comment (chuanhtuan) [inline app/views/words/show.html.erb]: Constant should be defined in model or Setting

File: app/assets/javascripts/application.js [hunk 3]
```diff
@@ -13,4 +13,6 @@
 //= require jquery
 //= require jquery_ujs
 //= require turbolinks
-//= require_tree .
+//= require bootstrap
+//= require_tree
```
Comment (dieunb) [inline app/assets/javascripts/application.js]: missing `.`

File: app/views/words/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<% provide(:title, t("words"))   %>
```
Comment (dieunb) [inline app/views/words/index.html.erb]: remove `()` the first

File: config/routes.rb [hunk 5]
```diff
@@ -4,13 +4,13 @@
...
   get "signup"  => "users#new"
-    resources :users
+  resources :users
 
   get "lessons" => "categories#index"
   resources :categories, only: [:show, :index]
 
-  get "words"   => "words#show"
-  resources :words, only: [:show]
+  get "words"   => "words#index"
```
Comment (dieunb) [inline config/routes.rb]: no need when you are using ` resources :words, only: [:index]`

File: app/views/users/_activity.html.erb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<% unless @activity.blank? %>
+<ul class="microposts">
+  <% @activity.each do |act| %>
+  <li>
```
Comment (dieunb) [inline app/views/users/_activity.html.erb]: indent

File: app/views/users/new.html.erb [hunk 7]
```diff
@@ -1,4 +1,4 @@
-<% provide(:title, t('signup')) %>
+<% provide :title, t('signup') %>
```
Comment (dieunb) [inline app/views/users/new.html.erb]: use double quote mark instead single

**Pull request merged**